gtsam/geometry/Line3.cpp

```diff
@@ -0,0 +1,118 @@
+#include <gtsam/geometry/Line3.h>
+
+namespace gtsam {
+
+Line3 Line3::retract(const Vector4 &v, OptionalJacobian<4, 4> H) const {
+    Vector3 w;
+    w << v[0], v[1], 0;
+    Rot3 eps;
+    if (H) {
+        Eigen::Matrix3d Dw_mat = Eigen::Matrix3d::Zero();
+        OptionalJacobian<3, 3> Dw(Dw_mat);
+        Dw->setZero();
+        eps = Rot3::Expmap(w, Dw);
+        H->block<2, 2>(0, 0) = Dw->block<2, 2>(0, 0);
+        (*H)(2, 2) = 1;
```

Comment:
I could not do H->topLeft<2,2>(), because Eigen::Map has no member topLeft. 
I also tried `eps = Rot3::Expmap(w, H->block<2, 2>(0, 0));`. It complains that there's no viable conversion from Eigen::block to OptionalJacobian. 
Apart from this, doesn't Rot3::Expmap expect Matrix3 for the Jacobian argument? 
What I am trying to do is use the <2,2> topLeft of this Matrix3. Would it be possible to directly pass a <2,2> matrix or block to Rot3::Expmap? 
So I'm considering doing this: 
`Matrix3 Dw;
eps = Rot3::Expmap(w, Dw);
H->setIdentity();
H->block<2, 2>(0, 0) = Dw.block<2, 2>(0, 0);`

Reply:
Yes, that works. Ps, when I see `eps` I think scalar. I propose you use `incR`.